tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,76 @@
+import functools
+import inspect
+from unittest import mock
+import os
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pytest
+
+import mlflow
+from mlflow.utils import (
+    truncate_dict,
+)
+from mlflow.utils.mlflow_tags import MLFLOW_PARENT_RUN_ID, MLFLOW_AUTOLOGGING
+from mlflow.utils.validation import (
+    MAX_PARAMS_TAGS_PER_BATCH,
+    MAX_METRICS_PER_BATCH,
+    MAX_PARAM_VAL_LENGTH,
+    MAX_ENTITY_KEY_LENGTH,
+)
+
+from tests.autologging.fixtures import test_mode_off
+
+
+from tests.spark_autologging.utils import spark_session  # pylint: disable=unused-import
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.regression import LinearRegression
+from mlflow.pyspark.ml import _get_estimator_param_map
+
+MODEL_DIR = "model"
+
+
+def truncate_param_dict(d):
+    return truncate_dict(d, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
+
+
+def get_expected_class_tags(estimator):
+    return {
+        'estimator_name': estimator.__class__.__name__,
+        'estimator_class': estimator.__class__.__module__ + "." + estimator.__class__.__name__,
+    }
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+    return data.params, data.metrics, tags, artifacts
```

Comment:
I originally defined this function but maybe we should store the result in a named tuple to make it easier to access each attribute like:

```python
run_data = get_run_data(...)

assert run_data.params = ...
assert run_data.artifacts = ...
```